proto/catalog.proto

```diff
@@ -7,18 +7,24 @@ import "plan_common.proto";
 
 option optimize_for = SPEED;
 
+message ColumnIndex {
+  uint64 index = 1;
+}
+
 message StreamSourceInfo {
   map<string, string> properties = 1;
   plan_common.RowFormatType row_format = 2;
   string row_schema_location = 3;
-  int32 row_id_index = 4;
+  ColumnIndex row_id_index = 4;
   repeated plan_common.ColumnCatalog columns = 5;
   repeated int32 pk_column_ids = 6;
 }
 
 message TableSourceInfo {
-  repeated plan_common.ColumnCatalog columns = 1;
-  map<string, string> properties = 2;
+  ColumnIndex row_id_index = 1;
```

Comment:
maybe this field should be in the `SourceExecutor` node? (we can do it later.) c.c. @BugenZhao

Reply:
Yes. Currently we assemble a `ProstSource` and then convert it to `SourceCatalog` to construct the `StreamSource` node, which is weird. We can refactor it later. 🥰